src/index.ts

```diff
@@ -857,8 +857,12 @@ export async function processMessage<
   id: ConversationId,
   encryptedMessage: string
 ): Promise<DecodedMessage<ContentTypes>> {
-  const json = XMTPModule.processMessage(client.inboxId, id, encryptedMessage)
-  return DecodedMessage.from(json, client)
+  const obj = await XMTPModule.processMessage(
+    client.inboxId,
+    id,
+    encryptedMessage
+  )
+  return DecodedMessage.fromObject(obj, client)
```

Comment:
```suggestion
  return DecodedMessage.from(json, client)
```

Reply:
The thing returned from XmtpModule is json not a object you have to do from here like we do in other places in the index.

https://github.com/xmtp/xmtp-react-native/blob/a955447c1eff205418be567e75a0369859d3bc7c/src/index.ts#L351-L359

Reply:
This was failing with json parse error while my code was working! Logging the json variable showed it was an object not a string?

Reply:
It 100% returns a string https://github.com/xmtp/xmtp-react-native/blob/main/ios/XMTPModule.swift#L1309.